Mathlib/Data/Nat/Fib/Basic.lean

```diff
@@ -150,6 +164,21 @@ theorem fib_add (m n : ℕ) : fib (m + n + 1) = fib m * fib n + fib (m + 1) * fi
     simp only [fib_add_two, ih]
     ring
 
+lemma fib_add_ne_zero {m n : ℕ} (hn : n ≠ 0) :
+    fib (m + n) = fib m * fib (n - 1) + fib (m + 1) * fib n := by
+  obtain ⟨i, rfl⟩ := Nat.exists_eq_add_one_of_ne_zero hn
+  simp [← add_assoc, fib_add]
+
+lemma fib_add_sub_one {m n : ℕ} (hn : n ≠ 0) :
+    fib (m + n - 1) = fib m * (fib n - fib (n - 1))
+      + fib (m + 1) * fib (n - 1) := by
+  rw [fib_sub_one (by omega), fib_add, fib_add_ne_zero hn, Nat.sub_add_eq]
+  nth_rw 3 [fib_sub_one hn]
+  simp [Nat.mul_sub]
+  rw [← Nat.sub_add_eq, tsub_add_tsub_comm
+    (Nat.mul_le_mul (le_refl _) (fib_le_fib_of_le (by omega)))
```

Comment:
I think you can golf this proof slightly by using [`mul_left_mono`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Monoid/Unbundled/Basic.html#mul_left_mono).